subprojects/base-services/src/main/java/org/gradle/internal/exceptions/NonGradleCauseException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.exceptions;
+
+/**
+ * An exception that is not caused by a Gradle specific problem.
+ * e.g. a problem caused by a compilation error.
+ * This serves as a marker interface.
+ */
```

Comment:
💄 Although used line withs are inconsistent on other classes' Javadoc, most are 120, 140 or 200 chars wide. We should use one of the formats.